src/containerapp/azext_containerapp/custom.py

```diff
@@ -2096,7 +2096,7 @@ def set_registry(cmd, name, resource_group_name, server, username=None, password
     registries_def = None
     registry = None
 
-    if "registries" not in containerapp_def["properties"]["configuration"]:
+    if not containerapp_def["properties"]["configuration"].get("registries"):
```

Comment:
Should we just use `safe_get(containerapp_def, "properties", "configuration", "registries", default=[])`?

Reply:
We can but then we would have to update the `containerapp_def["properties"]["configuration"]["registries"]` to point to registries_def. Otherwise in the default case, you would update the new empty list that got returned, but the `containerapp_def` would be unchanged